Code refactoring, duplicate card entry accepted, proper file handling, better file/paste importing

arthurauffray opened the pull request:
1. 
Refactored code for better modularity and readability. (functions)
Improved exception handling for user inputs and currency conversions.
Fixed issue where duplicate card entries did not update correctly.
Enhanced handling of card data to ensure proper grouping and display.
Used context managers for file operations to ensure proper handling.

2.
file imports now in CSV format
paste accepts same format as file import

arthurauffray committed:
main.py improvements

Refactored code for better modularity and readability. (functions)
Improved exception handling for user inputs and currency conversions.
Fixed issue where duplicate card entries did not update correctly.

Enhanced handling of card data to ensure proper grouping and display.
Used context managers for file operations to ensure proper handling.

arthurauffray committed:
paste/file imports, better input handling
file imports now in CSV format
paste accepts same format as file import

arthurauffray commented:
updated gitignore too

arthurauffray reviewed (commented):
all tests passing

F1nn1125 commented:
I'll take a look sometime today

F1nn1125 commented:
I appreciate the time you took to refactor the code, I know it is probably horribly formatted and written.

MTGA deck lists are always stored as plain text (at least from what I've seen). They are imported directly from your clipboard into MTGA,

I designed the input system to allow users to easily copy and paste decks from any MTGA deck list website. Unfortunately, your pull request breaks this ease of use by changing the way users input card names and prices. (I'm assuming you're not familiar with MTGA deck formatting.)

I don't think there's really a use for csv on importing files, however, I can see a use for the export to be a csv.

arthurauffray committed:
patched for MTGA file import format

- pasting only in mtga format
- export to csv or txt
- better, uniform input looping
